skbio/metadata/tests/test_interval.py

```diff
@@ -446,6 +446,34 @@ def test_init_upper_bound_lt_lower_bound(self):
         with self.assertRaises(ValueError):
             IntervalMetadata(-1)
 
+    def test_upper_bound_is_none(self):
```

Comment:
In those tests, the private attributes have to be used any way and the code is longer to create expected objects, but I agree with you that it'd be better to use `__eq__` in case the internal attributes change in the future. I reorganize and modify the tests to accommodate your request.